samples/openapi3/client/petstore/java/jersey2-java8/src/test/java/org/openapitools/client/ApiClientTest.java

```diff
@@ -0,0 +1,64 @@
+package org.openapitools.client;
+
+import org.openapitools.client.model.*;
+import org.openapitools.client.ApiClient;
+
+import java.lang.Exception;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.TimeZone;
+import java.text.SimpleDateFormat;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import org.junit.*;
+import static org.junit.Assert.*;
+
+public class ApiClientTest {
+    ApiClient apiClient = null;
+    Pet pet = null;
+
+    @Before
+    public void setup() {
+        apiClient = new ApiClient();
+        pet = new Pet();
+    }
+
+    @Test
+    public void testSerializeToString() throws Exception {
+        Long petId = 4321L;
+        pet.setId(petId);
+        pet.setName("jersey2 java8 pet");
+        Category category = new Category();
+        category.setId(petId);
+        category.setName("jersey2 java8 category");
+        pet.setCategory(category);
+        pet.setStatus(Pet.StatusEnum.AVAILABLE);
+        pet.setPhotoUrls(Arrays.asList("A", "B", "C"));
+        Tag tag = new Tag();
+        tag.setId(petId);
+        tag.setName("jersey2 java8 tag");
+        pet.setTags(Arrays.asList(tag));
+
+        String result = "{\"id\":4321,\"category\":{\"id\":4321,\"name\":\"jersey2 java8 category\"},\"name\":\"jersey2 java8 pet\",\"photoUrls\":[\"A\",\"B\",\"C\"],\"tags\":[{\"id\":4321,\"name\":\"jersey2 java8 tag\"}],\"status\":\"available\"}";
+        assertEquals(result, apiClient.serializeToString(pet, null, "application/json", false));
+        // nulllable and there should be no diffencne as the payload is not null
+        assertEquals(result, apiClient.serializeToString(pet, null, "application/json", true));
+
+        // non-nullable null object should give "" (empty body)
```

Comment:
I think this use case is non trivial to understand because there may be two variants:
1. The request is supposed to have an empty body by design, i.e. the OpenAPI doc request or response schema indicates the body is empty.
1. The client or server is setting the data incorrectly. I.e. the OpenAPI doc indicates the object is not nullable, but the actually either the client or server are setting to a null value.

Is it even possible?

Reply:
> The client or server is setting the data incorrectly

This can be prevented by marking the "body" parameter (OAS v2 term) as required.